cmd/root.go

```diff
@@ -163,6 +165,62 @@ func parseToolsFile(ctx context.Context, raw []byte) (ToolsFile, error) {
 	return toolsFile, nil
 }
 
+func parse(ctx context.Context, logger log.Logger) {
+	logger.DebugContext(ctx, "Attempting to parse updated tools file.")
+	// TODO: add logic for parsing
+}
+
+// watchFile checks for changes in the provided yaml tools file.
+func watchFile(toolsFileName string, ctx context.Context, logger log.Logger) {
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		logger.WarnContext(ctx, "error setting up new watcher %s", err)
+	}
+
+	defer w.Close()
+
+	err = w.Add(filepath.Dir(toolsFileName))
+	if err != nil {
+		logger.WarnContext(ctx, "error adding the tools file to watcher %s", err)
+	}
+
+	logger.InfoContext(ctx, fmt.Sprintf("Now watching tools file %s", toolsFileName))
+	var debounceTimer *time.Timer
+	debounceDelay := 100 * time.Millisecond
+	for {
+		select {
+		case <-ctx.Done():
+			logger.WarnContext(ctx, "watcher context cancelled")
+			return
+		case err, ok := <-w.Errors:
+			if !ok {
+				logger.WarnContext(ctx, "error watcher alredy closed %s", err)
+			}
+
+			if err != nil {
+				logger.WarnContext(ctx, "error watching file %s", err)
+			}
+		case e, ok := <-w.Events:
+			if !ok {
+				logger.WarnContext(ctx, "error with event %s", err)
```

Comment:
Will this exit if there's error?
```suggestion
				logger.WarnContext(ctx, "file watcher error with event: %s", err)
```